src/wallet/wallet.cpp

```diff
@@ -3936,9 +3936,11 @@ CKeyPool::CKeyPool(const CPubKey& vchPubKeyIn, bool internalIn)
 
 int CWalletTx::GetDepthInMainChain(interfaces::Chain::Lock& locked_chain) const
 {
+    assert(pwallet != nullptr);
+    AssertLockHeld(pwallet->cs_wallet);
     if (isUnconfirmed() || isAbandoned()) return 0;
 
-    return locked_chain.getBlockDepth(m_confirm.hashBlock) * (isConflicted() ? -1 : 1);
+    return (pwallet->GetLastBlockHeight() - m_confirm.block_height + 1) * (isConflicted() ? -1 : 1);
```

Comment:
0ff03871add000f8b4d8f82aeb168eed2fc9dc5f

I don't think commit message is correct:
```
Avoid to lock chain to query state thanks to tracking last block
height in CWallet.
```
Should say something like?
```
Use wallet's last block height instead of the locked chain, which
will allow removing the chain lock.
```

Reply:
That's a better wording, I will use it if I have to rebase.